litestar/contrib/minijnja.py

```diff
@@ -130,7 +160,9 @@ def get_template(self, template_name: str) -> MiniJinjaTemplate:
         except MiniJinjaTemplateNotFound as exc:
             raise TemplateNotFoundException(template_name=template_name) from exc
 
-    def register_template_callable(self, key: str, template_callable: Callable[[dict[str, Any]], Any]) -> None:
+    def register_template_callable(
+        self, key: str, template_callable: TemplateCallableType[StateProtocol, P, T]
+    ) -> None:
         """Register a callable on the template engine.
```

Comment:
I believe the typing of this method was incorrect before these changes. Reason being that `template_callable` is constrained to receive a `dict` but we decorate with `pass_state` which means it will receive minijinja's `State` object.